controllers/deactivation/deactivation_controller.go

```diff
@@ -164,6 +196,21 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	if !states.Deactivating(usersignup) {
 		states.SetDeactivating(usersignup, true)
 
+		// Before we update the UserSignup in order to set the deactivating state, we should reset the scheduled
+		// deactivation time if required just in case the current value is nil or has somehow changed.  Since the UserSignup
+		// controller is going to be reconciling immediately after setting the deactivating state then it will be
+		// creating a deactivating notification, meaning that the scheduled deactivation time that we set here is going
+		// to be extremely temporary (as it will be recalculated after the next reconcile), however it is done for correctness
+		scheduledDeactivationTime := (*mur.Status.ProvisionedTime).Add(deactivationTimeout)
```

Comment:
Lines 204-210 seem to be 100% the same as lines 178-186. So it's a good candidate to move to a function.

Reply:
Done, this code has been refactored into a separate function.

Reply:
It looks like you forgot to commit it. I don't see any changes :)

Reply:
oops, committed